src/main/kotlin/io/github/zap/build/gradle/convention/ProjectExtension.kt

```diff
@@ -169,4 +171,99 @@ fun Project.configureShadow(destination: File,
     tasks.named("build", DefaultTask::class.java) {
         it.dependsOn(sj)
     }
+}
+
+fun Project.qs (name: String = "", options: ExternalModuleDependency.() -> Unit = {}) : ExternalModuleDependency.() -> Unit {
+    return {
+        val actualName = name.ifEmpty { this.name }
+        version {
+            if (isLocalTag(actualName)) {
+                it.require(getLocalVersion(actualName))
+            } else {
+                if(version != null && version!!.isNotEmpty()) {
+                    ext["qs-$actualName-ver"] = version
```

Comment:
I don't quite understand the purpose of setting the ext here
from what I see it is only used by getVerFromAncestor which in turn is only used if the version is null or empty 
if it's null or empty than this ext variable would never have been set in the first place
so why are we setting the variable

Reply:
subproject